src/extension.ts

```diff
@@ -104,115 +99,111 @@ function getProjectPrefix(): string[] {
+        files = files.filter((file) => extname(file.toLowerCase()) === ".sqf");
 
-        // Convert fnc_function_name.sqf -> PREP(function_name);
-        files.forEach((file, index) => {
-            let functionName = file.split(".")[0]; // Remove extension
+        files = files.map(file => {
+            let functionName = parse(file).name; // Remove extension
             functionName = (functionName.split("_").splice(1)).join("_"); // Remove fn_ / fnc_ prefix
-            files[index] = `PREP(${functionName});`;
-        }, files);
+            return `PREP(${functionName});`;
+        });
 
-        // Filter out files that didn't match
-        files = files.filter(file => file !== "PREP(undefined);");
         const content = files.join("\n");
 
-        logMessage("TRACE", `content=${content}`);
+        logMessage(ELogLevel.TRACE, `content=${content}`);
         files.sort();
 
         functionsFolderArray.pop(); // Remove "functions", XEH_PREP should be in addon root
-        const prepFileDir = functionsFolderArray.join("\\") + "\\XEH_PREP.hpp";
-        fs.writeFile(prepFileDir, content, err => {
-            if (err) {
-                vscode.window.showErrorMessage(`Failed to create file at ${prepFileDir}`);
-            } else {
-                vscode.window.showInformationMessage(`Generated XEH_PREP.hpp file for ${files.length} functions`);
-            }
-        });
+        const prepFileDir = join(...functionsFolderArray, "XEH_PREP.hpp");
+        try {
+            await fs.writeFile(prepFileDir, content);
+            await vscode.window.showInformationMessage(`Generated XEH_PREP.hpp file for ${files.length} functions`);
+        } catch (err) {
+            await vscode.window.showErrorMessage(`Failed to create file at ${prepFileDir}`);
+        }
     });
     context.subscriptions.push(generatePrepFile);
 
-    const generateStringtableKey = vscode.commands.registerTextEditorCommand("lazyarmadev.generateStringtableKey", function (textEditor: vscode.TextEditor) {
+    const generateStringtableKey = vscode.commands.registerTextEditorCommand("lazyarmadev.generateStringtableKey", async (textEditor: vscode.TextEditor) =>  {
+        if (!textEditor) { return; }
         const document = textEditor.document;
         const match = document!.fileName.match(addonDiskRegex);
 
         const stringtableDir = `${match![0]}\\stringtable.xml`;
-        logMessage("TRACE", `stringtableDir=${stringtableDir}`);
+        logMessage(ELogLevel.TRACE, `stringtableDir=${stringtableDir}`);
 
-        const [, prefix, component] = getProjectPrefix();
+        const projectPrefix = await getProjectPrefix();
         let stringKey = document!.getText(document!.getWordRangeAtPosition(textEditor!.selection.active));
-        stringKey = `STR_${prefix}_${component}_${stringKey}`;
-        logMessage("TRACE", `stringKey="${stringKey}"`);
+        stringKey = `STR_${projectPrefix.mainPrefix}_${projectPrefix.component}_${stringKey}`;
```

Comment:
Should use prefix, not main prefix
```suggestion
        stringKey = `STR_${projectPrefix.prefix}_${projectPrefix.component}_${stringKey}`;
```